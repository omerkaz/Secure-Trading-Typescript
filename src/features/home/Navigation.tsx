import { Link } from 'react-router-dom';

function Navigation() {
    return (
        <div className="row">
            <div className="row justify-content-center mb-2">
                <div className="col-xs-6 col-md-6">
                    <Link className="btn btn-dark w-100" to={'sellerapplication'}>
                        Satıcı Başvurusu Oluştur
                    </Link>
                </div>
            </div>
            <div className="row justify-content-center mb-2">
                <div className="col-xs-6 col-md-6">
                    <Link className="btn btn-dark w-100" to={'buyerapplication'}>
                        Alıcı Başvurusu Oluştur
                    </Link>
                </div>
            </div>
            <div className="row justify-content-center mb-2">
                <div className="col-xs-6 col-md-6">
                    <Link className="btn btn-dark w-100" to={'applicationstatus'}>
                        Başvuru Durum Sorgula
                    </Link>
                </div>
            </div>
            <div className="row justify-content-center">
                <div className="col-xs-6 col-md-6">
                    <Link className="btn btn-dark w-100" to={'applicationcancel'}>
                        Başvuru İptal
                    </Link>
                </div>
            </div>
        </div>
    );
}

export default Navigation;
