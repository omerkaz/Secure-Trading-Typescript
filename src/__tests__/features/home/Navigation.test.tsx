import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Navigation from '@features/home/Navigation';

describe('Navigation elements renders correctly', () => {
    test('snapshot test', () => {
        const componentTree = render(
            <BrowserRouter>
                <Navigation />
            </BrowserRouter>,
        );
        expect(componentTree).toMatchSnapshot();
    });

    test('elements test', () => {
        render(
            <BrowserRouter>
                <Navigation />
            </BrowserRouter>,
        );
        const link1 = screen.getByText('Satıcı Başvurusu Oluştur');
        expect(link1).toBeInTheDocument();
        const link2 = screen.getByText('Alıcı Başvurusu Oluştur');
        expect(link2).toBeInTheDocument();
        const link3 = screen.getByText('Başvuru Durum Sorgula');
        expect(link3).toBeInTheDocument();
        const link4 = screen.getByText('Başvuru İptal');
        expect(link4).toBeInTheDocument();
    });
});
