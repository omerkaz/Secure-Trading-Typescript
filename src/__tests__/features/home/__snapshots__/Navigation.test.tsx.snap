// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation elements renders correctly snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <div
          class="row justify-content-center mb-2"
        >
          <div
            class="col-xs-6 col-md-6"
          >
            <a
              class="btn btn-dark w-100"
              href="/sellerapplication"
            >
              Satıcı Başvurusu Oluştur
            </a>
          </div>
        </div>
        <div
          class="row justify-content-center mb-2"
        >
          <div
            class="col-xs-6 col-md-6"
          >
            <a
              class="btn btn-dark w-100"
              href="/buyerapplication"
            >
              Alıcı Başvurusu Oluştur
            </a>
          </div>
        </div>
        <div
          class="row justify-content-center mb-2"
        >
          <div
            class="col-xs-6 col-md-6"
          >
            <a
              class="btn btn-dark w-100"
              href="/applicationstatus"
            >
              Başvuru Durum Sorgula
            </a>
          </div>
        </div>
        <div
          class="row justify-content-center"
        >
          <div
            class="col-xs-6 col-md-6"
          >
            <a
              class="btn btn-dark w-100"
              href="/applicationcancel"
            >
              Başvuru İptal
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <div
        class="row justify-content-center mb-2"
      >
        <div
          class="col-xs-6 col-md-6"
        >
          <a
            class="btn btn-dark w-100"
            href="/sellerapplication"
          >
            Satıcı Başvurusu Oluştur
          </a>
        </div>
      </div>
      <div
        class="row justify-content-center mb-2"
      >
        <div
          class="col-xs-6 col-md-6"
        >
          <a
            class="btn btn-dark w-100"
            href="/buyerapplication"
          >
            Alıcı Başvurusu Oluştur
          </a>
        </div>
      </div>
      <div
        class="row justify-content-center mb-2"
      >
        <div
          class="col-xs-6 col-md-6"
        >
          <a
            class="btn btn-dark w-100"
            href="/applicationstatus"
          >
            Başvuru Durum Sorgula
          </a>
        </div>
      </div>
      <div
        class="row justify-content-center"
      >
        <div
          class="col-xs-6 col-md-6"
        >
          <a
            class="btn btn-dark w-100"
            href="/applicationcancel"
          >
            Başvuru İptal
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
